UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -187,31 +188,63 @@ public async Task<YearWeekResponseDataModel> GetAllYearWeek(string correlationId
+                BatchSearchResponse searchResult = new();
+                bool isCached = false;
+                const string frequency = "daily";
+                const string rowKey = "DailyKey";
 
-                string searchText = $" and $batch(Frequency) eq 'Daily' and $batch(Content) eq null ";
-                if (_userService.IsDistributorUser)
+                if (_cacheConfiguration.Value.IsFssCacheEnabled)
                 {
-                    searchText = $" and $batch(Frequency) eq 'Daily' and $batch(Content) eq 'Tracings' ";
+                    BatchSearchResponseModel batchSearchResponseModel = await _fileShareServiceCache.GetBatchResponseFromCache(PartitionKey, rowKey, frequency, correlationId);
+
+                    if (batchSearchResponseModel.BatchSearchResponse != null)
+                    {
+                        searchResult = batchSearchResponseModel.BatchSearchResponse;
+                        isCached = true;
+                    }
                 }
 
-                IFileShareApiClient fileShareApiClient = new FileShareApiClient(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+                if (searchResult.Entries == null)
+                {
+                    string accessToken = await _authFssTokenProvider.GenerateADAccessToken(_userService.IsDistributorUser, correlationId);
 
-                IResult<BatchSearchResponse> result = await _fileShareService.FSSBatchSearchAsync(searchText, accessToken, correlationId, fileShareApiClient);
+                    _logger.LogInformation(EventIds.ShowDailyFilesResponseStarted.ToEventId(), "Maritime safety information request to get daily NM files response started for daily user:{SignInName} and Identity:{userId} with _X-Correlation-ID:{correlationId}", _userService.SignInName ?? "Public", _userService.UserIdentifier, correlationId);
 
-                BatchSearchResponse searchResult = result.Data;
+                    string searchText = $" and $batch(Frequency) eq 'Daily' and $batch(Content) eq null ";
+                    if (_userService.IsDistributorUser)
+                    {
+                        searchText = $" and $batch(Frequency) eq 'Daily' and $batch(Content) eq 'Tracings' ";
+                    }
+
+                    IFileShareApiClient fileShareApiClient = new FileShareApiClient(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
+
+                    IResult<BatchSearchResponse> result = await _fileShareService.FSSBatchSearchAsync(searchText, accessToken, correlationId, fileShareApiClient);
+
+                    searchResult = result.Data;
+
+                    if (_cacheConfiguration.Value.IsFssCacheEnabled && searchResult != null && searchResult.Entries.Count > 0)
```

Comment:
Please update if conditions to Weekly & Batch Attribute methods similar to this.

Reply:
done Changes